tests/conftest.py

```diff
@@ -289,9 +289,14 @@ def compare_post(data: str, entity: str = entity) -> Any:
 
         legacy_data = json.loads(legacy_resp.data)
         snql_data = json.loads(snql_resp.data)
-        assert (
-            legacy_data["sql"] == snql_data["sql"]
-        ), f"LEGACY:\n{legacy_data['sql']}\n\nSNQL:\n{snql_data['sql']}\n"
+
+        if legacy_data.get("sql"):
+            assert (
+                legacy_data["sql"] == snql_data["sql"]
+            ), f"LEGACY:\n{legacy_data['sql']}\n\nSNQL:\n{snql_data['sql']}\n"
+        else:
+            # There was a validation error, the response should be identical
```

Comment:
We want to be able to compare queries that return a validation error, not just those that succeed. In that case there is no key `sql` in the response

Reply:
I didn't think that the error outputs would be the same, since the error messages might end up being different. If this works though, then that's great!